src/apps/statements/service/storeStatements/updateFullActivities.ts

```diff
@@ -44,78 +78,84 @@ export default async ({ config, models, client }: Opts): Promise<void> => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.extensions !== undefined
-      ) ? matchingActivity.definition.extensions : {};
-    });
-    const types = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.type !== undefined
-      ) ? matchingActivity.definition.type : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-    const moreInfos = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.moreInfo !== undefined
-      ) ? matchingActivity.definition.moreInfo : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-
-    return {
-      activityId,
-      name: Object.assign({}, ...names),
-      description: Object.assign({}, ...descriptions),
-      extensions: Object.assign({}, ...extensions),
-      ...(types.length > 0 ? { type: types[types.length - 1] } : {}),
-      ...(moreInfos.length > 0 ? { moreInfo: moreInfos[moreInfos.length - 1] } : {}),
-    };
-  });
-
-  await config.repo.updateFullActivities({ updates, client });
-  // tslint:disable-next-line:max-file-line-count
+  const groupedActivities = groupBy(
+    definedActivities,
+    (activity) => activity.activityId,
+  );
+
+  const fullActivities = map(
+    groupedActivities,
+    (matchingActivities, activityId): FullActivityModel => {
+      const name = last(
+        matchingActivities
+          .map((matchingActivity) => matchingActivity.name)
+          .filter((value) => value !== undefined),
+      );
+
+      const description = last(
+        matchingActivities
+          .map((matchingActivity) => matchingActivity.description)
+          .filter((value) => value !== undefined),
+      );
+
+      const extensions = last(
+        matchingActivities
+          .map((matchingActivity) => matchingActivity.extensions)
+          .filter((value) => value !== undefined),
+        );
+
+      const contextActivities = last(
+        matchingActivities
+        .map((matchingActivity) => matchingActivity.contextActivities)
+        .filter((value) => value !== undefined),
+      );
+
+      const type = last(
+        matchingActivities
+        .map((matchingActivity) => matchingActivity.type)
+        .filter((value) => value !== undefined),
+      );
+
+      const moreInfo = last(
+        matchingActivities
+        .map((matchingActivity) => matchingActivity.moreInfo)
+        .filter((value) => value !== undefined),
+      );
+
+      return {
+        activityId,
+        lrsId: client.lrs_id,
+        organisationId: client.organisation,
+        ...(name !== undefined ? { name } : {}),
+        ...(description !== undefined ? { description } : {}),
+        ...(extensions !== undefined ? { extensions } : {}),
```

Comment:
@PrinceWaune the tests are currently failing because the logic for these properties has changed when it didn't need to be changed. For example, for the `name` property, we need to use `Object.assign({}, ...names)` as before and `names` should be an array not a single item as it is now using the `last` function.